src/Methods/Macro.php

```diff
@@ -58,15 +62,27 @@ final class Macro implements BuiltinMethodReflection
      *
      * @param string $className
      * @param string $methodName
-     * @param \ReflectionFunction $reflectionFunction
+     * @param ReflectionFunction $reflectionFunction
      */
-    public function __construct(string $className, string $methodName, \ReflectionFunction $reflectionFunction)
+    public function __construct(string $className, string $methodName, ReflectionFunction $reflectionFunction)
     {
         $this->className = $className;
         $this->methodName = $methodName;
         $this->reflectionFunction = $reflectionFunction;
         $this->parameters = $this->reflectionFunction->getParameters();
         $this->isStatic = false;
+
+        if ($this->reflectionFunction->isClosure()) {
+            try {
+                /** @var Closure $closure */
+                $closure = $this->reflectionFunction->getClosure();
+                Closure::bind($closure, new stdClass);
```

Comment:
@kylekatarnls You could use `if ($closure instanceof Closure)` here

Reply:
I should not, I **know** `$closure instanceof Closure` is always true. It's a PHPStan mistake because it cannot detect that `getClosure()` cannot be `null` if `isClosure()` is true. I should just do `Closure::bind($this->reflectionFunction->getClosure(), new stdClass);` it would be the correct way. I just add the typing to chill PHPStan analyse wrong error `Parameter #1 $old of static method Closure::bind() expects Closure, Closure|null given.`

Reply:
PHPStan cannot see the relation between `isClosure()` and `getClosure()` return type without your help.

Reply:
I know, it's fair. What do you mean by "help"? The typing I added or is there an other way you recommend?